src/UglyToad.PdfPig/Content/Pages.cs

```diff
@@ -79,6 +106,53 @@ internal Page GetPage(int pageNumber, NamedDestinations namedDestinations, Parsi
             return page;
         }
 
+        internal void AddPageFactory<TPage>(IPageFactory<TPage> pageFactory)
+        {
+            Type type = typeof(TPage);
+            if (pageFactoryCache.ContainsKey(type))
+            {
+                throw new InvalidOperationException($"Could not add page factory for page type '{type}' as it was already added.");
+            }
+
+            pageFactoryCache.Add(type, pageFactory);
+        }
+
+        internal void AddPageFactory<TPage, TPageFactory>() where TPageFactory : IPageFactory<TPage>
+        {
+            var constructor = typeof(TPageFactory).GetConstructor(new[]
+            {
+                typeof(IPdfTokenScanner),
+                typeof(IResourceStore),
+                typeof(ILookupFilterProvider),
+                typeof(IPageContentParser),
+                typeof(ParsingOptions)
+            });
+
+            if (constructor == null)
+            {
+                throw new InvalidOperationException($"Could not find valid constructor for page factory of type '{typeof(TPageFactory)}'. " +
+                                                    "The page factory should have a constructor with the following parameters: " +
+                                                    $"{typeof(IPdfTokenScanner)}, {typeof(IResourceStore)}, {typeof(ILookupFilterProvider)}, {typeof(IPageContentParser)}, {typeof(ParsingOptions)}.");
+            }
+
+            var instance = constructor.Invoke(new object[]

```

Comment:
Use of reflection may be problematic if we look to support AOT builds in future but will do for now.

Reply:
it will definitely be a problem as methods using reflection might be trimmed out.

I have started working on .Net 8 AOT support, here is an example of fix:
https://github.com/BobLd/PdfPig/commit/79f41038c06158b4d8cee319ed179d1f0296d0fa#diff-b302975a4cc11925011fc218ef0f880d103758850dd9eddc73b5f61968335b8cR208-R209

I will work on on adding full AOT support soon, happy for someone else to work on it / help though